tests/unit/test_utils.py

```diff
@@ -493,6 +493,104 @@ def test_tell_and_seek(self):
         self.assertEqual(chunk.tell(), 3)
         chunk.seek(0)
         self.assertEqual(chunk.tell(), 0)
+        chunk.seek(1, whence=1)
+        self.assertEqual(chunk.tell(), 1)
+        chunk.seek(-1, whence=1)
+        self.assertEqual(chunk.tell(), 0)
+        chunk.seek(-1, whence=2)
+        self.assertEqual(chunk.tell(), 2)
+
+    def test_tell_and_seek_boundaries(self):
+        data = b'thisstringis28characterslong'
+        start_pos = 10
+        chunk_size = 8
+
+        # create test file
+        filename = os.path.join(self.tempdir, 'foo')
+        with open(filename, 'wb') as f:
+            f.write(data)
+
+        chunk = ReadFileChunk.from_filename(
+            filename, start_byte=start_pos, chunk_size=chunk_size)
+        byte_buffer = io.BytesIO(data[start_pos:start_pos+chunk_size])
+
+        for obj in (chunk, byte_buffer):
+            self._assert_whence_start_behavior(obj)
+            self._assert_whence_end_behavior(obj)
+            self._assert_whence_relative_behavior(obj)
+            self._assert_boundary_behavior(obj)
+
```

Comment:
It would also be great to add assertions to make sure the progress callbacks are reporting the correct amount as well based on the various values of `whence` we provide.